### PR TITLE
Implementing all the function in the scm-base class

### PR DESCRIPTION
**Pull request by shruthi-venkateswaran**
Moving most of the code from models

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
+'use strict';
+const Breaker = require('circuit-fuses');
+const Github = require('github');
+const schema = require('screwdriver-data-schema');
```
Comment (jer) [inline index.js]: Is this the only lower-case one on purpose?

Comment (nkatzman) [inline index.js]: it's the only one that's not a class?

**Comment by jer** [comment]
👍

**Comment by d2lam** [comment]
LGTM 👍

**Pull request merged**